lib/semantics/check-select-stmt.cc

```diff
@@ -0,0 +1,103 @@
+//= check-select-stmt.cc - Checker for select-case, select-rank, select-type ==
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//=--------------------------------------------------------------------------==
+
+#include "check-select-stmt.h"
+#include "tools.h"
+#include "../parser/message.h"
+
+namespace Fortran::semantics {
+
+void SelectStmtChecker::Leave(
+    const parser::CaseConstruct &selectcaseConstruct) {
+  const auto &selectcaseStmt{
+      std::get<parser::Statement<parser::SelectCaseStmt>>(
+          selectcaseConstruct.t)};
+  const auto &parsedExpr{
+      std::get<parser::Scalar<parser::Expr>>(selectcaseStmt.statement.t).thing};
+  TypeCategory selectcasestmttype;
```

Comment:
The style used in the rest of the project is to capitalize words after the first in local variable names. So `selectCaseStmtType` rather than `selectcasestmttype`, etc.

Reply:
Will do this change..

Reply:
Also for the function argument.